projects/iio_demo_freeRTOS/src/platform/maxim/main.c

```diff
@@ -0,0 +1,143 @@
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include "platform_includes.h"
+#include "common_data.h"
+
+#ifdef IIO_EXAMPLE
+#include "iio_example.h"
+#endif
+
+/**
+ * @brief IIOD example task
+ * @return Returns the error code of iio_example_main function
+*/
+int iiodTask()
+{
+	return iio_example_main();
+}
+
+/**
+ * @brief LED blinking task for multithreading example
+ * @return Returns error code
+*/
+int blinkingTask()
+{
+	int ret = 0;
+	struct no_os_gpio_desc *led_pin;
+	ret = no_os_gpio_get(&led_pin, &gpio_ip);
+	if (ret)
+		return ret;
+
+	ret = no_os_gpio_direction_output(led_pin, NO_OS_GPIO_LOW);
+	if (ret)
+		return ret;
+
+	bool led_on = true;
+	while (1) {
+		if (!led_on) {
+			ret = no_os_gpio_set_value(led_pin, NO_OS_GPIO_HIGH);
+			if (ret)
+				break;
+			led_on=true;
+			no_os_mdelay(100);
+		}
+		if (led_on) {
+			ret = no_os_gpio_set_value(led_pin, NO_OS_GPIO_LOW);
+			if (ret)
+				break;
+			led_on=false;
+			no_os_mdelay(100);
+		}
+	}
+	return ret;
+}
+
+/**
+ * @brief Creating iio example with blinking led task
+ * @return  Returns error code
+ *
+*/
+int create_tasks(void)
+{
+	int ret;
+	for (int i = 0; i < 0x00FFFF; i++) {}
```

Comment:
it's not all that clear why this for is needed. maybe a comment would be useful. also, if this is intended to be a delay, maybe it would be better to use no_os_delay

Reply:
removed, not needed